### PR TITLE
api: Add routes for accessing active community

### PR DESCRIPTION
**Pull request by kulla**
Add the routes `/api/user/active-authors` and `/api/user/active-authors` for accessing the ids for active authors / reviewers.

### Testing
Links:

* http://en.serlo.localhost:4567/api/user/active-reviewers
* http://en.serlo.localhost:4567/api/user/active-authors

 You can change the config [`$mysql_timestamp_for_active_community`](https://github.com/serlo/serlo.org/blob/378f0711909505164a7982b4ab1a8d7fb99a02d4/packages/public/server/src/config/definitions.local.php#L63) for changing the end date of the 90 days period in which the edits / reviews are calculated. You can execute `yarn mysql:import-anonymous-data` and change this value to `CURDATE()` to test the behavior in production. The current values should be similar to

* https://github.com/serlo/serlo.org/blob/master/packages/public/client/src/feature-prototypes/donation-profiles/active-reviewers.ts
* https://github.com/serlo/serlo.org/blob/master/packages/public/client/src/feature-prototypes/donation-profiles/active-authors.ts

### CODE REVIEW

File: packages/public/server/src/module/Api/src/Factory/UserApiControllerFactory.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    protected function createController(ServiceLocatorInterface $serviceManager)
+    {
+        /** @var AuthorizationService $authorizationService */
+        $authorizationService = $serviceManager->get(
+            AuthorizationService::class
+        );
+        $controller = new UserApiController($authorizationService);
+
+        /** @var UserManager $userManager */
```
Comment (inyono) [inline packages/public/server/src/module/Api/src/Factory/UserApiControllerFactory.php:42]:
```suggestion
        /** @var UserManagerInterface $userManager */
```
+ Import

File: packages/public/server/README.md [hunk 2]
```diff
@@ -106,3 +106,10 @@ Covers some best practices that new code should comply to.
   There is a possibility to define taxonomy terms in which each edit shall be automatically reviewed.
   This is useful when you want for example set up a sandbox area in which all edits shall be automatically online.
   To archieve this you need to set the configuration variable `autoreview_taxonomy_term_ids` to the list of those taxonomy term ids.
+- **Timestamp for calculating the active community:**
+  There are two routes `/api/user/active-authors` and `/api/user/active-reviewers` for accessing the active authors or active reviewers which are defined by 10 edits / reviews in the last 90 days.
+  However in the local database there are no edits in the last years.
+  Thus when we calculate the active community in the local development environment both events would be empty.
+  So we introduced the configuration `mysql_timestamp_for_active_community`.
+  It is the mysql endpoint of the time frame in which the active community is calculated.
+  In production set it to `CURDATE()`, for development you can something like `Date("2014-01-01")`.
```
Comment (inyono) [inline packages/public/server/README.md:115]: Actually: can you make this optional so that we don't need to change anything in production?

Comment (kulla) [reply packages/public/server/README.md:115]: `CURDATE()` is already the default value (see https://github.com/serlo/serlo.org/blob/092ef8ea49a4397a1b9be401422185af50a9ad58/packages/public/server/src/module/User/src/Factory/UserManagerFactory.php#L61 ). I have updated the README.

**Comment by kulla** [comment]
@inyono Ready to be merged :-)

File: packages/public/server/README.md (hunk 2 above)
Comment (inyono) [reply packages/public/server/README.md]: This still needs work, unfortunately. When you remove the value from `definitions.local.php`, PHP  fails with:
```
Fatal error: Uncaught ErrorException: Undefined variable: mysql_timestamp_for_active_community in /usr/local/apache2/htdocs/src/config/autoload/global.php:160 Stack trace: #0 /usr/local/apache2/htdocs/src/config/autoload/global.php(160): exception_error_handler(8, 'Undefined varia...', '/usr/local/apac...', 160, Array) #1 /usr/local/apache2/htdocs/src/vendor/zendframework/zendframework/library/Zend/Config/Factory.php(109): include('/usr/local/apac...') #2 /usr/local/apache2/htdocs/src/vendor/zendframework/zendframework/library/Zend/ModuleManager/Listener/ConfigListener.php(376): Zend\Config\Factory::fromFile('config/autoload...') #3 /usr/local/apache2/htdocs/src/vendor/zendframework/zendframework/library/Zend/ModuleManager/Listener/ConfigListener.php(144): Zend\ModuleManager\Listener\ConfigListener->addConfigByPath('config/autoload...', 'glob_path') #4 [internal function]: Zend\ModuleManager\Listener\ConfigListener->onMergeConfig(Object(Zend\ModuleManager\ModuleEvent)) #5 /usr/local/apache2/htdocs/src/vendor/zendframework in /usr/local/apache2/htdocs/src/config/autoload/global.php on line 160
```
which means we need to change `definitions.local.php` in production (breaking change). If you catch that error (e.g. using `isset`), we can continue without any breaking changes.

Comment (kulla) [reply packages/public/server/README.md]: Fixed in https://github.com/serlo/serlo.org/pull/476/commits/6ad6d1b73420251d0ef5574402c32373bc0fdb72

**Comment by kulla** [comment]
@inyono new attempt :-)

**Pull request merged**